src/operator/rnn_impl.h

```diff
@@ -127,9 +127,9 @@ void LstmForwardTraining(DType* ws,
                          bool state_outputs,
                          const int L,
                          const int D,
```

Comment:
What are D and L ? can D*L be >5B ?

Reply:
L is `num_layers`, and D is `direction`. I believe we agreed that we would not support > 2**32 layers, so L should be fine as an `int`. D can have two possible values, 0 or 1, to indicate whether to run the op with bidirectional recurrent layers. Consequently, D*L can never be >5B.

Reply:
fair to keep D and L as int then.